### DISCUSSION TITLE
Save uncompressed PDF

### DISCUSSION THREAD

**Discussion by Andreasox** [pymupdf/PyMuPDF#2990] [Q&A]
Is it possible to create a totally uncompressed PDF?
Testing with 
doc.save(__file__.replace(".py", ".pdf"),deflate=False,expand=255, garbage=2, pretty=True)
creates the attached PDF with, what I understand, all objects created with draw_rect, insert_text etc. included as binary streams.
[scratch3.pdf](https://github.com/pymupdf/PyMuPDF/files/13859486/scratch3.pdf)

**Comment by JorjMcKie** [comment]
Should work like that. Any complaints?

**Comment by Andreasox** [comment]
We were discussing if there is an "easy way" to group objects as described in https://github.com/pymupdf/PyMuPDF/issues/2973 by search/replace. Guess we are too old-fashioned and used to EPS text files ...
